File: Cpp/Penyimpanan.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
using namespace std;

// Base class untuk penyimpanan (implementasi hierarchical inheritance)
class Penyimpanan : public Komponen {
protected:
    int kapasitasGB;
    string interface; // SATA, NVMe, etc.

public:
    Penyimpanan() : Komponen() {
        this->kapasitasGB = 0;
        this->interface = "";
    }

    Penyimpanan(int kapasitasGB, string interface, string merk, string nama, int harga) 
        : Komponen(merk, nama, harga) {
        this->kapasitasGB = kapasitasGB;
        this->interface = interface;
    }

    void setKapasitasGB(int kapasitasGB) {
        this->kapasitasGB = kapasitasGB;
    }

    void setInterface(string interface) {
        this->interface = interface;
    }

    int getKapasitasGB() const {
        return this->kapasitasGB;
    }

    string getInterface() const {
        return this->interface;
    }

    void displayInfo() const override {
        Komponen::displayInfo();
        cout << "Kapasitas: " << this->kapasitasGB << " GB" << endl;
        cout << "Interface: " << this->interface << endl;
    }

    ~Penyimpanan() {}
};

File: Java/Penyimpanan.java
// Base class untuk penyimpanan
public class Penyimpanan extends Komponen {
    protected int kapasitasGB;
    protected String interface_; // SATA, NVMe, etc.

    public Penyimpanan() {
        super();
        this.kapasitasGB = 0;
        this.interface_ = "";
    }

    public Penyimpanan(int kapasitasGB, String interface_, String merk, String nama, int harga) {
        super(merk, nama, harga);
        this.kapasitasGB = kapasitasGB;
        this.interface_ = interface_;
    }

    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }

    public void setInterface(String interface_) {
        this.interface_ = interface_;
    }

    public int getKapasitasGB() {
        return this.kapasitasGB;
    }

    public String getInterface() {
        return this.interface_;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Kapasitas: " + this.kapasitasGB + " GB");
        System.out.println("Interface: " + this.interface_);
    }
}
